pkg/cluster/space/openebs.go

```diff
@@ -0,0 +1,599 @@
+func (o *OpenEBSDiskSpaceValidator) parseFstabContainerOutput(output []byte) ([]string, error) {
+	seen := map[string]bool{}
+	mounts := []string{}
+	buf := bytes.NewBuffer(output)
+	scanner := bufio.NewScanner(buf)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "#") {
+			continue
+		}
+
+		words := strings.Fields(line)
+		if len(words) < 2 || !strings.HasPrefix(words[1], "/") {
+			continue
+		}
+
+		if _, ok := seen[words[1]]; ok {
+			continue
+		}
+
+		seen[words[1]] = true
+		mounts = append(mounts, words[1])
+	}
+	if len(mounts) == 0 {
+		return nil, fmt.Errorf("failed to locate any mount point")
+	}
+	return mounts, nil
+}
+
+// buildTmpPVC creates a temporary PVC requesting for 1Mi of space.
+func (o *OpenEBSDiskSpaceValidator) buildTmpPVC(node string) *corev1.PersistentVolumeClaim {
+	tmp := uuid.New().String()[:5]
+	pvcName := fmt.Sprintf("disk-free-%s-%s", node, tmp)
+	if len(pvcName) > 63 {
+		pvcName = pvcName[0:31] + pvcName[len(pvcName)-32:]
+	}
+
+	return &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pvcName,
+			Namespace: "default",
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			StorageClassName: pointer.String(o.dstSC),
+			AccessModes:      []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceStorage: resource.MustParse("1Mi"),
+				},
+			},
+		},
+	}
+}
+
+// OpenEBSVolume represents an OpenEBS volume in a node. Holds space related information and
+// a flag indicating if the volume is part of the root (/) volume.
+type OpenEBSVolume struct {
+	Free       int64
+	Used       int64
+	RootVolume bool
+}
+
+// deleteTmpPVCs deletes the provided pvcs from the default namespace and waits until all their
+// backing pvs dissapear as well (this is mandatory so we don't leave any orphan pv as this would
+// make the pvmigrate to fail). this function has a timeout of 5 minutes, after that an error is
+// returned.
+func (o *OpenEBSDiskSpaceValidator) deleteTmpPVCs(ctx context.Context, pvcs []*corev1.PersistentVolumeClaim) error {
+	pvs, err := o.kcli.CoreV1().PersistentVolumes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list persistent volumes: %w", err)
+	}
+
+	pvsByPVCName := map[string]corev1.PersistentVolume{}
+	for _, pv := range pvs.Items {
+		if pv.Spec.ClaimRef == nil {
+			continue
+		}
+		pvsByPVCName[pv.Spec.ClaimRef.Name] = pv
+	}
+
+	for _, pvc := range pvcs {
+		propagation := metav1.DeletePropagationForeground
+		delopts := metav1.DeleteOptions{PropagationPolicy: &propagation}
+		if err := o.kcli.CoreV1().PersistentVolumeClaims("default").Delete(
+			ctx, pvc.Name, delopts,
+		); err != nil {
+			if errors.IsNotFound(err) {
+				continue
+			}
+			return fmt.Errorf("failed to delete temp pvc %s: %w", pvc.Name, err)
```

Comment:
should failing to delete one pvc fail all? im pretty sensitive to tasks that dont do a good job of cleaning up after themselves

Reply:
fixed.